arex-instrumentation-api/src/main/java/io/arex/inst/runtime/util/ReplayUtil.java

```diff
@@ -0,0 +1,127 @@
+import io.arex.inst.runtime.serializer.Serializer;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Predicate;
+
+/**
+ * Replay all mockers under case
+ * (called once after arexContext init)
+ */
+public class ReplayUtil {
+
+    /**
+     * init replay all mocker under case and cached replay result
+     */
+    public static void queryMockers() {
+        if (!ContextManager.needReplay()) {
+            return;
+        }
+        try {
+            QueryAllMockerDTO requestMocker = new QueryAllMockerDTO();
+            requestMocker.setRecordId(ContextManager.currentContext().getCaseId());
+            requestMocker.setCategoryTypes(new String[]{MockCategoryType.DYNAMIC_CLASS.getName(), MockCategoryType.REDIS.getName()});
+            requestMocker.setReplayId(ContextManager.currentContext().getReplayId());
+            List<Mocker> allMockerList = MockUtils.queryMockers(requestMocker);
+            if (CollectionUtil.isEmpty(allMockerList)) {
+                return;
+            }
+
+            filterMergeMocker(allMockerList);
+
+            Map<Integer, List<Mocker>> cachedReplayResultMap = ContextManager.currentContext().getCachedReplayResultMap();
+            cachedReplayResultMap.clear();
+            buildReplayResultMap(allMockerList, cachedReplayResultMap);
+
+            ascendingSortByCreationTime(cachedReplayResultMap);
+        } catch (Exception e) {
+            LogManager.warn("replay.all.mocker", e);
+        }
+    }
+
+    /**
+     * compatible with merge record, after batchSave published can be removed
+     */
+    private static void filterMergeMocker(List<Mocker> allMockerList) {
+        List<Mocker> splitMockerList = new ArrayList<>();
+        Predicate<Mocker> filterMergeRecord = mocker -> ArexConstants.MERGE_RECORD_NAME.equals(mocker.getOperationName());
+        for (Mocker mergeMocker : allMockerList) {
+            if (!filterMergeRecord.test(mergeMocker)) {
+                continue;
+            }
+            List<MergeDTO> mergeReplayList = Serializer.deserialize(mergeMocker.getTargetResponse().getBody(), ArexConstants.MERGE_TYPE);
+            if (CollectionUtil.isEmpty(mergeReplayList)) {
+                continue;
+            }
+            splitMockerList.addAll(convertMergeMocker(mergeReplayList));
+        }
+        if (CollectionUtil.isEmpty(splitMockerList)) {
+            return;
+        }
+        allMockerList.removeIf(filterMergeRecord);
+        allMockerList.addAll(splitMockerList);
+    }
+
+    private static List<Mocker> convertMergeMocker(List<MergeDTO> mergeReplayList) {
+        List<Mocker> convertMockerList = new ArrayList<>();
+        for (MergeDTO mergeDTO : mergeReplayList) {
+            if (mergeDTO == null || (!MockCategoryType.DYNAMIC_CLASS.getName().equals(mergeDTO.getCategory())
+                    && !MockCategoryType.REDIS.getName().equals(mergeDTO.getCategory()))) {
```

Comment:
Support for merged categories needs to be configurable and get in one place.

Reply:
has declare predicate, temporary compatible with merge record, after batchSave published can be removed.